lib/cmd/fh3/build.js

```diff
@@ -0,0 +1,330 @@
+    if (!params.config) {
+      params.config = 'debug';
+    } else if (params.config === 'release' || params.config === 'distribution') {
+      if (!params.privateKeyPass) {
+        return cb(i18n._("Missing 'privateKeyPass' parameter"));
+      }
+      if (!params.certpass) {
+        return cb(i18n._("Missing 'certpass' parameter"));
+      }
+    }
+    if (isIosBuild(params)) {
+      params.deviceType = params.destination;
+      if (!params.bundleId) {
+        return cb(i18n._("Missing 'bundleId' parameter"));
+      }
+    }
+    // set default build version if none specified. Note this is hardcoded as not available from millicore API.
+    // note these values were gotten from studio/static/js/application/destinations/destination_*
+    if (!params.version) {
+      if (params.destination === 'android') {
+        params.version = '4.0';
+      } else if (isIosBuild(params)) {
+        params.version = '6.0';
+      }
+    }
+    return cb(null, params);
+  },
+  'customCmd' : function(params,cb) {
+    try {
+      mutateGitRefArgs(params);
+      doBuild(params, cb);
+    } catch (x) {
+      log.silly(x);
+      return cb(i18n._("Error processing params: ") + x + "\n");
+    }
+  }
+};
+
+/**
+ * Check if the destination is iphone, ipad or ios
+ * @param params
+ * @returns {boolean}
+ */
+function isIosBuild(params) {
+  return params.destination === "iphone" || params.destination === "ipad" || params.destination === "ios";
+}
+
+/**
+ * Mutates the arguments object in order to support --git-branch --git-commit and --git-tag, turning them into gitRef
+ * @param  {Object} params Arguments object
+ */
+function mutateGitRefArgs(params) {
+  // Workaround for current millicore behavior:
+  // If no git- params are present, build from branch:master
+  var gitParams = [
+    'git-commit',
+    'git-tag',
+    'git-branch'
+  ];
+  if (!_.any(gitParams, _.propertyOf(params))) {
+    params.gitRef = {
+      type: 'branch',
+      value: 'master'
+    };
+    return;
+  }
+
+  function clearGitParams(params) {
+    _.each(gitParams, function(p) {
+      delete params[p];
+    });
+  }
+
+  // priority is commit > tag > branch
+  if (params['git-commit']) {
+    params.gitRef = {
+      type: 'commit',
+      value: params['git-commit'],
+      hash: params['git-commit']
+    };
+    clearGitParams(params);
+    return;
+  }
+
+  if (params['git-tag']) {
+    params.gitRef = {
+      type: 'tag',
+      value: params['git-tag']
+    };
+    clearGitParams(params);
+    return;
+  }
+
+  if (params['git-branch']) {
+    params.gitRef = {
+      type: 'branch',
+      value: params['git-branch']
+    };
+    clearGitParams(params);
+    return;
+  }
+}
+
+/**
+ * Perform the build
+ * @param params
+ * @param cb
+ */
+function doBuild(params, cb) {
+  var uri = "box/srv/1.1/wid/" + fhc.curTarget + "/" + params.destination + "/" + params.app + "/deliver";
+  var message = "";
+  var doCall = function() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Function has a complexity of 24 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)